### PR TITLE
[Feature] [Style] Add Spotless plugin to fix formatting and style errors automatically

### LINKED ISSUE #2492: [Feature][Style] Use Spotless to fix formatting and style errors automatically
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

* Use `Spotless` maven plugin to detect and fix formatting and style errors automatically.
* I will send an email to dev mail list for detailed discussions later.

### Usage Scenario

* Currently, ST uses `Checkstyle` to detect formatting and style errors. However, it will be much more conveniently if we could find those style errors and automatically fix them before pushing the code.
* We could introduce `spotless` plugin into DS and use it to automatically fix the style together with `Pre-commit Hook`.
* https://github.com/diffplug/spotless
* https://github.com/diffplug/spotless/tree/main/plugin-maven

### Related issues

* We have already done this in the Apache DolphinScheduler project, see https://github.com/apache/dolphinscheduler/issues/10963 for details.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EricGao888**
## Purpose of this pull request

* Replaced `Checkstyle` and `Scala Style` with `Spotless` to fix formatting and style errors automatically
* Gave examples on style fix of `.java, .md, .scala` files.
* Updated related docs.
* This PR closes: #2492 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by EricJoy2048** [comment]
This is a very helpful feature for us. @531651225  @legendtkl  @TyrantLucifer  @laglangyue  @getChan  @2013650523  @TaoZex  @liugddx  PTAL.

File: tools/style/eclipse.importorder [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#Organize Import Order
+0=org.apache.seatunnel
+1=org.apache
+2=java
+3=javax
+4=org
+5=com
```
Comment (ashulin) [inline tools/style/eclipse.importorder]: Can scala be added?
And change the file name to `seatunnel.importorder`

Comment (EricGao888) [reply tools/style/eclipse.importorder]: Thanks for the suggestions. I will take a look and see whether we could add scala into it.

**Comment by hailin0** [comment]
great

**Comment by EricGao888** [comment]
@ashulin Sorry for the late update, I add extra configurations for scala formatting configurations and rename the previous `eclipse.importorder` to `seatunnel.importorder`. BTW, I found there are lot of suppressions related to checkstyle, I suppose I need to remove them as well?

https://github.com/apache/incubator-seatunnel/blob/dev/tools/checkstyle/suppressions.xml
https://github.com/apache/incubator-seatunnel/blob/c29370a7b5bf6d70e4266b4b4b345e059a699148/seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java#L51

**Comment by EricJoy2048** [comment]
@ashulin  PTAL

**Comment by FlechazoW** [comment]
Hi, @EricGao888 Is there any process here? If this is not convenient for you, I can continue to do this.

**Comment by EricGao888** [comment]
> Hi, @EricGao888 Is there any process here? If this is not convenient for you, I can continue to do this.

Hello @FlechazoW , thanks for reaching out. Recently I've been quite busy with something else. I would appreciated it if you could help continue to complete this work. BTW,  the configuration of `ratchetFrom` in this PR may not function as expected. You may refer to https://github.com/diffplug/spotless/issues/710#issuecomment-1211495549 for a workaround. If possible, I suggest not using `ratchetFrom` but fixing the legacy formatting errors once for all.

**Comment by FlechazoW** [comment]
> > Hi, @EricGao888 Is there any process here? If this is not convenient for you, I can continue to do this.
> 
> Hello @FlechazoW , thanks for reaching out. Recently I've been quite busy with something else. I would appreciated it if you could help continue to complete this work. BTW, the configuration of `ratchetFrom` in this PR may not function as expected. You may refer to [diffplug/spotless#710 (comment)](https://github.com/diffplug/spotless/issues/710#issuecomment-1211495549) for a workaround. If possible, I suggest not using `ratchetFrom` and fix the legacy formatting errors once for all.

Thanks for your advice, and I will perfect this as much as possible.

**Comment by EricGao888** [comment]
Closed due to https://github.com/apache/incubator-seatunnel/pull/2641#issuecomment-1379913743

**Pull request closed without merging**